src/screens/MessageScreen.tsx

```diff
@@ -1,90 +1,116 @@
+      />
+    ),
+    [Edit2, navigation, tileId],
+  );
 
   useLayoutEffect(() => {
     navigation.setOptions({
-      title: t('messages-title', 'My Messages'),
+      title: t('messages-title', 'Messages'),
+      headerRight: iconButton,
     });
-  }, [navigation]);
+  }, [iconButton, navigation]);
 
   const { styles } = useStyles(defaultStyles);
-  const { userDetailsList, fetchNextPage, hasNextPage, isLoading } =
-    useMyMessages(tileId);
 
   const handlePostTapped = useCallback(
-    (userId: string | undefined, displayName?: string) => () => {
-      if (userId) {
-        navigation.navigate('Home/DirectMessage', {
-          recipientUserId: userId,
-          displayName,
-        });
-      }
+    (tappedUsers: UserProfile[], conversationId: string) => () => {
+      navigation.navigate('Home/DirectMessage', {
+        users: tappedUsers,
+        conversationId,
+      });
     },
     [navigation],
   );
 
   const renderLeft = useCallback(
-    (props: { style: ViewStyle }, user: User) => (
-      <View
-        style={{
-          flexDirection: 'row',
-          justifyContent: 'center',
-          marginLeft: 8,
-        }}
-      >
-        {
-          <Badge
-            size={10}
-            style={[
-              styles.badgeView,
-              user.isUnread
-                ? styles.badgeColor?.enabled
-                : styles.badgeColor?.disabled,
-            ]}
-            testID={tID('unread-badge')}
-          />
-        }
-        <View
-          style={
-            user.isUnread
-              ? [props.style, styles.newMessageIconView]
-              : props.style
-          }
-        >
-          <GiftedAvatar user={user} textStyle={{ fontWeight: '500' }} />
-        </View>
-      </View>
-    ),
-    [
-      styles.badgeColor?.disabled,
-      styles.badgeColor?.enabled,
-      styles.badgeView,
-      styles.newMessageIconView,
-    ],
+    (selectedProfiles: UserProfile[], hasUnread: boolean) => {
+      if (selectedProfiles && selectedProfiles.length > 0) {
+        return (
+          <View
+            style={{
+              flexDirection: 'row',
+              justifyContent: 'center',
+              marginLeft: 8,
+            }}
+          >
+            {
+              <Badge
+                size={10}
+                style={[
+                  styles.badgeView,
+                  hasUnread
+                    ? styles.badgeColor?.enabled
+                    : styles.badgeColor?.disabled,
+                ]}
+                testID={
+                  hasUnread ? tID('unread-badge') : tID('unread-badge-hidden')
+                }
+              />
+            }
+            <View>
+              {
+                // TODO: Combine multiple GiftedAvatars
+              }
+              <GiftedAvatar
+                key={selectedProfiles[0].id}
+                user={{
+                  name: selectedProfiles[0].profile.displayName,
```

Comment:
For simplicity we just show the icon for the first user in the group but would be nice to handle multiple icons here.